frontend/src/vault/qr/upload/useProcessQrMutation.ts

```diff
@@ -55,42 +52,22 @@ export const useProcessQrMutation = () => {
       }
 
       const url = code.data;
-      console.log('url:', url);
       const queryParams = getRawQueryParams<QrQueryParams>(url);
       const { jsonData } = queryParams;
       const payload = await decompressQrPayload(jsonData);
 
       if ('type' in queryParams) {
-        return match(queryParams.type, {
+        const path = await match(queryParams.type, {
           NewVault: async () => {
-            match(queryParams.tssType, {
-              Keygen: async () => {
-                const keygenMsg = KeygenMessage.fromBinary(payload);
-                queryClient.setQueryData(
-                  ['keygenMessage', keygenMsg.sessionId],
-                  keygenMsg
-                );
-
-                navigate(
-                  makeAppPath('joinKeygen', {
-                    keygenType: queryParams.tssType,
-                    sessionID: keygenMsg.sessionId,
-                  })
-                );
-              },
-              Reshare: async () => {
-                const reshareMsg = ReshareMessage.fromBinary(payload);
-                queryClient.setQueryData(
-                  ['reshareMessage', reshareMsg.sessionId],
-                  reshareMsg
-                );
-                navigate(
-                  makeAppPath('joinKeygen', {
-                    keygenType: queryParams.tssType,
-                    sessionID: reshareMsg.sessionId,
-                  })
-                );
-              },
+            const keygenType = queryParams.tssType;
+
+            const { fromBinary } = keygenMsgRecord[keygenType];
+
+            const keygenMsg = JSON.stringify(fromBinary(payload).toJson());
+
+            return makeAppPath('joinKeygen', {
```

Comment:
this , is it mean `keygenMsg` is passed as a query string to the next view?

Reply:
Correct, react-query shouldn't be used to pass state between pages.

E.g. in the "Send" example. The send button that navigates to the "Send page" doesn't have to pass balances to the "Send page", since "Send page" can simply query them on it's own with react-query.